modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileFeatureSource.java

```diff
@@ -599,4 +609,69 @@ protected boolean handleVisitor(Query query, FeatureVisitor visitor) throws IOEx
         return super.handleVisitor(query, visitor);
     }
 
+    @Override
+    protected List<String> getNativeStyles() {
+        List<String> list = new ArrayList<>();
+        if (this.shpFiles.isLocal()) {
+            if (this.shpFiles.exists(SLD)) {
+                list.add(this.shpFiles.getTypeName());
+            }
+        }
+        return list;
+    }
+
+    @Override
+    protected StyledLayerDescriptor getDefaultStyle() {
+        StyledLayerDescriptor result = null;
+        try {
+            result = getNativeStyle("");
+        } catch (Exception e) {
+            // do nothing
+        }
+        return result;
+    }
+
+    @Override
+    protected StyledLayerDescriptor getNativeStyle(String name) throws IOException {
+        ReadableByteChannel in = null;
+        try {
+            ByteBuffer buffer = ByteBuffer.allocate(4*1024);
+            FileReader reader = new FileReader() {
+                public String id() {
+                    return "Shapefile's getStyle method to read sidecar style file";
+                }
+            };
+            in = this.shpFiles.getReadChannel(SLD, reader);
```

Comment:
Calling directly shpFiles.getInputStream for this case it's probably easier? :-)
(and then give the stream to Parser directly)